### ISSUE TITLE
[Bug] AddOn 'PremadeGroupsFilter' tried to call the protected function 'GetPlaystyleString()'.

### ISSUE THREAD

**Issue by tryallthethings** [0xbs/premade-groups-filter#89]
Receiving a lot of errors from the addon in Bugsack:

```
93x [ADDON_ACTION_BLOCKED] AddOn 'PremadeGroupsFilter' tried to call the protected function 'GetPlaystyleString()'.
[string "@!BugGrabber\BugGrabber.lua"]:519: in function <!BugGrabber\BugGrabber.lua:519>
[string "=[C]"]: in function `GetPlaystyleString'
[string "@FrameXML\LFGList.lua"]:3508: in function <FrameXML\LFGList.lua:3499>
[string "=[C]"]: ?
[string "=[C]"]: in function `LFGListUtil_SetSearchEntryTooltip'
[string "@FrameXML\LFGList.lua"]:2576: in function <FrameXML\LFGList.lua:2573>
[string "=[C]"]: ?
```

Potential duplicate of #88, although my account is fully secured:

![image](https://user-images.githubusercontent.com/11146296/163675374-9265ced7-d613-41a5-b8a3-b2eddbc27d1c.png)

Running latest version of Premade Groups Filter (2.8.2).

**Comment by 0xbs** [comment]
Very interesting. Are you able to create premade mythic plus groups with custom arbitrary titles and comments, especially a Mists of Tirna Scithe group? I am testing against the mists dungeon, maybe you are not authenticated for that dungeon for another reason like not being character level 60 or something else.

**Comment by tryallthethings** [comment]
![image](https://user-images.githubusercontent.com/11146296/163677941-acb672e9-57d8-4a3b-88ec-2d071932d98f.png)

No issues there. Also no LUA errors in Bugsack.

**Comment by tryallthethings** [comment]
One thing I forgot to mention: I'm not having any issues with Premade Groups Filter, I just noticed the LUA errors in Bugsack. So I can't really tell you when the issues occurred or know how to reproduce them.

**Comment by 0xbs** [comment]
Could you please type `/dump C_LFGList.IsPlayerAuthenticatedForLFG(703)` into your chat window and see if it returns true or false? This checks if you are allowed to create groups with arbitrary titles for mists on mythic.

Also do you have a keystone for mists in your bags? This could change some behavior on Blizzards side.

If the above function returns true, my code should apply a fix for the BugSack. So maybe another addon is doing something strange. In this case please do another test with all addons disabled except PGF. BugSack usually appears whenever you hover over a group to see the tooltip.

If it returns false, I do not apply a fix because I assume that you cannot use custom titles. At least that's what I understand from reading Blizzard's interface code. This would explain the BugSacks, but not why you are allowed to create groups with custom titles.

**Comment by tryallthethings** [comment]
> Thank you. The error/BugSack should be reproducible by hovering over an entry in group list filtered by PGF. The tooltip triggers it, because it tries to show the play style in the second line of the tooltip which fails if PGF removes groups from the list and hence taints the code because GetPlaystyleString is protected.

Yes, that works. But only once after starting the game and logging into the character.

> Could you please type `/dump C_LFGList.IsPlayerAuthenticatedForLFG(703)` into your chat window and see if it returns true or false? This checks if you are allowed to create groups with arbitrary titles for mists on mythic plus.

Returns true for me.

> Also do you have a keystone for mists in your bags? This could change some behavior on Blizzards side.

No, I currently have Tazavesh Streets and Court of Stars in my inventory. 

> If the above function returns true, my code should apply a fix for the BugSack. So maybe another addon is doing something strange. In this case please do another test with all addons disabled except PGF. Simply apply a PGF filter and hover over a group.
 
The error also appears with only Buggrabber, Bugsack and PGF active. I tried this on another char with no keystone in the inventory.

**Comment by cremor** [comment]
I can confirm this problem. My account is also fully secured and I can create custom groups. But that error is thrown and I can't see the play style in tooltips.
The dump command posted above also returns true for me.

I've noticed that the error is **not** thrown again after an UI reload. Play style is also shown fine after a UI reload. Only a full relog causes the error again.

**Comment by 0xbs** [comment]
This is very helpful. I assume that when the addon is initalized after a relog, the information if an account is fully secured is not yet fully available and the function briefly returns false.

On a later reload (not relog) the data is already in the cache and hence everything works as expected.

To fix the issue, I have to somehow postpone the call to `IsPlayerAuthenticatedForLFG()`.

**Comment by tryallthethings** [comment]
> On a later reload (not relog) the data is already in the cache and hence everything works as expected.
> 
> To fix the issue, I have to somehow postpone the call to `IsPlayerAuthenticatedForLFG()`.

I doubt that this will fix the issue. Even after being logged in for a while (> 10 minutes) the error still appears. What I (and probably @cremor too) meant, was that the error only appears once after logging into a character, opening the group finder with a PGF filter active (partyfit in my case) and hovering over any listed key. It doesn't appear each time you open the group finder. Only after you switch to a different character or return to the character selection screen + log into the same character again, the issue will appear again **once**.

**Comment by cremor** [comment]
@tryallthethings No, @0xbs understood me correctly. It doesn't matter how long you wait to open the LFG list. The addon checks the value of that function immediately on login.

**Comment by 0xbs** [comment]
If ADDON_LOADED is a sufficient delay for you guys, I will implemented the fix as suggested and release a new version tomorrow if my time schedule works out. Still strange that I never encountered the issue myself, but could be different on other regions or realms.

**Comment by cremor** [comment]
I havent tested it myself yet. But isn't ADDON_LOADED still very early in the login process? Maybe PLAYER_LOGIN would be a better event?

Also, isn't ADDON_LOADED fired for each addon that is loaded? So the code posted by @rodneycheung would be executed multiple times.

**Comment by 0xbs** [comment]
Yes, PLAYER_LOGIN is better and the code above is executed for every addon. I already considered that. Just give me some time until my actual work finishes and I will do the changes.

**Comment by 0xbs** [comment]
Resolved in version 2.8.3

**Comment by tryallthethings** [comment]
Looks good. Can't reproduce the error anymore. 😄 Thank you!

**Comment by Iowerth** [comment]
Hi. I read FAQ, but - what if I have no possibility to secure my account because of Blizzard's restrictions to my country? I don't need automatic group titles, I just need absence of this error.

**Comment by 0xbs** [comment]
Please open a new issue instead of commenting on very old ones.

You could remove or comment out line 35 in FixGetPlaystyleString.lua: https://github.com/0xbs/premade-groups-filter/blob/0309e8475d586d39d384b8920d251fecfa4d7312/Fixes/FixGetPlaystyleString.lua#L35
